CONTRIBUTING.md

```diff
@@ -31,19 +103,77 @@ Code contributions should be submitted in the form of a pull request. Here are t
 ```
 git clone <the url of your fork>
 ```
-Which branch should you base your work on? See [below](#branches) for a detailed answer (the non-detailed answer is the latest `rc/x.y.z` branch")
-* Start working! Make sure to tag the issue in your commits by including `#issue_number` in the commit messages. This will help make the larger purpose of your changes clear. 
-Also, every time you push to your fork, the issue page will be updated with a link to the commit, so that others can see the progress of your work.
-* **If you are adding functionality, you must also add tests.**
-* Run the tests! You can do this by running `make test` or `python -m pytest <a test file or directory>`. If you use the command in the Makefile, an html coverage report will be generated in the directory `htmlcov`.
-* When you are ready, create a pull request from your fork to the main repository. The title should be brief and descriptive, 
-and the body of the pull request should include the phrase `Relates to: #issue_number`, as well as a short description of the changes that you have implemented.
-    * Before creating a pull request, make sure that your changes do not conflict with work in the main repository by [rebasing or merging](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) from the main repository into your fork.
-
-###### Branches
-
-Historically, we have recommended basing work on either the `master` or `internal` branches.
-**We are now moving towards a more frequent and open release model and highly recommend basing all work off of the latest `rc/x.y.z` branch.**
-We plan on merging the latest `rc/x.y.z` branch into `master` roughly every 2 weeks.
-
-One reasonable heuristic for determining the latest `rc/x.y.z` branch and its estimated release date is to go to the [AllenSDK pull requests page](https://github.com/AllenInstitute/AllenSDK/pulls) and find the pull request titled `rc/x.y.z`. The pull request description text will describe a tentative release ("merge") date.
+* Make a new branch for your code off of `master`. For consistency and use with 
+visual git plugins, we prefer the following convention for branch naming:
+`GH-<issue-number>/<bugfix/feature>/<short-description>`. For example:
+    ```
+    GH-712/bugfix/auto-reward-key
+    GH-9999/feature/parallel-behavior-analysis
+    ```
+* Create an environment and install necessary requirements: `requirements.txt` and `test_requirements.txt`
+* Start writing code!
+
+### Style Guidelines
+We follow [PEP-8 guidelines](https://www.python.org/dev/peps/pep-0008/) for new python code.
+We also follow [PEP-484](https://www.python.org/dev/peps/pep-0484/) for type annotations.
```

Comment:
we do PEP-484?
we should update [this](http://confluence.corp.alleninstitute.org/display/IT/Informatics+dev+resources), or just change it to defer to this section of the style guidelines.

Reply:
Yes, I think we should just update that page to refer to these guidlines once they're finished. We've been including type annotations in all new code (that I've seen) which is covered by PEP 484.